pkg/spec/deployment.go

```diff
@@ -41,44 +40,10 @@ func (deployment *DeploymentSpecMod) Unmarshal(data []byte) error {
 	return nil
 }
 
-// TODO: abstract out this code when more controllers are added
 func (deployment *DeploymentSpecMod) Fix() error {
-
-	var err error
-
-	// fix deployment.Services
-	deployment.Services, err = fixServices(deployment.Services, deployment.Name)
-	if err != nil {
-		return errors.Wrap(err, "Unable to fix services")
-	}
-
-	// fix deployment.VolumeClaims
-	deployment.VolumeClaims, err = fixVolumeClaims(deployment.VolumeClaims, deployment.Name)
-	if err != nil {
-		return errors.Wrap(err, "Unable to fix persistentVolume")
-	}
-
-	// fix deployment.configMaps
-	deployment.ConfigMaps, err = fixConfigMaps(deployment.ConfigMaps, deployment.Name)
-	if err != nil {
-		return errors.Wrap(err, "unable to fix configMaps")
-	}
-
-	deployment.Containers, err = fixContainers(deployment.Containers, deployment.Name)
-	if err != nil {
-		return errors.Wrap(err, "unable to fix containers")
-	}
-
-	deployment.InitContainers, err = fixContainers(deployment.InitContainers, deployment.Name)
-	if err != nil {
-		return errors.Wrap(err, "unable to fix init-containers")
+	if err := deployment.ControllerFields.fixControllerFields(); err != nil {
```

Comment:
if this is the only thing happening in this function why not move everything from `fixControllerFields` here?

Reply:
because this same thing is being called in job.go in its Fix() method